org.jacoco.core.test/src/org/jacoco/core/internal/analysis/MethodAnalyzerTest.java

```diff
@@ -638,15 +641,84 @@ public void testTryCatchBlockFullyCovered() {
 		assertLine(1004, 0, 1, 0, 0);
 	}
 
-	private void runMethodAnalzer() {
+	// === Scenario: try/finally ===
+
+	private void createTryFinally() {
+		final Label l0 = new Label();
+		final Label l1 = new Label();
+		final Label l2 = new Label();
+		final Label l3 = new Label();
+
+		method.visitTryCatchBlock(l0, l2, l2, null);
+
+		method.visitLabel(l0);
+		method.visitLineNumber(1001, l0);
+
+		method.visitJumpInsn(Opcodes.IFEQ, l1);
+		// probe[0]
+		method.visitInsn(Opcodes.RETURN);
+		method.visitLabel(l1);
+		// probe[1]
+		method.visitInsn(Opcodes.RETURN);
+
+		method.visitLabel(l2);
+		method.visitJumpInsn(Opcodes.IFEQ, l3);
+		// probe[2]
+		method.visitInsn(Opcodes.RETURN);
+		method.visitLabel(l3);
+		// probe[3]
+		method.visitInsn(Opcodes.RETURN);
+	}
+
+	@Test
+	public void testTryFinallyWithoutFilter() {
+		createTryFinally();
+		probes[0] = true;
+		probes[3] = true;
+		runMethodAnalzer();
+		assertEquals(4, nextProbeId);
+
+		assertLine(1001, 2, 4, 2, 2);
+	}
+
+	private final IFilter FILTER = new IFilter() {
```

Comment:
Should this be a static constant? Or a instance member?

In any case I would prefer to name it specific for this test or make it a local variable in `testTryFinallyMergeSameBranch()`.

Reply:
> or make it a local variable in `testTryFinallyMergeSameBranch()`

it is also used in `testTryFinallyMergeDifferentBranches`

Reply:
Then `tryFinallyFilter`?

Reply:
Now `static` and `TRY_FINALLY_FILTER`. BTW note that it merges only branching instruction, but not whole duplicate.